src/lib.rs

```diff
@@ -644,6 +654,14 @@ impl<S: io::Read + io::Write> TlsStream<S> {
         Ok(self.0.tls_server_end_point()?)
     }
 
+    /// Returns the negotiated ALPN protocols
+    pub fn negotiated_alpn(&self) -> Result<Option<String>> {
+        Ok(self
+            .0
+            .negotiated_alpn()?
+            .map(|bytes| String::from_utf8(bytes).unwrap())) //negotiated_alpn always returns valid utf-8
```

Comment:
Is that really the case? I wouldn't be surprised if e.g. OpenSSL would just return whatever bytes the server sent.

Reply:
I indeed don't know this for sure not the implementation will use `from_utf8_lossy`